### PR TITLE
Update rubocop, rubocop-rspec, active_link_to, faker, slim_lint

### PR DESCRIPTION
**Pull request by slavakisel**
Noteworthy changes:

* **active_link_to**
  - Provide full url support
  - Allow `active: :exact` for exact matches

* **faker**
  - Lots of bug fixes -- most notably, a fix for email addresses and domains in non-en locales
  - Updated classes: Address, Avatar, Book, Code, Commerce, Company, Hipster, IDNumber, Internet, Number, Placeholdit, Shakespeare, and Time

* **rubocop-rspec**
  - Update to rubocop 0.39.0
  - Add `RSpec::NotToNot` and `Rspec/AnyInstance` cops

* **slim_lint**
  - Fix compatibility issues with Astrolabe gem by updating minimum RuboCop version to 0.36.0

* **rubocop**

Rubocop hasn’t been updated in almost a year. Since there were a huge amount of work done in versions 0.32 - 0.39 I’d like to mention just a few that are related to changes in rubocop.yml config and code:

  - New TargetRubyVersion configuration parameter can be used to specify which version of the Ruby interpreter the inspected code is intended to run on
  - Style/Documentation will exclude test directories by default
  - The RunRailsCops config parameter in .rubocop.yml is now obsolete. If someone attempts to use it, config validation will fail with a helpful message
  - New ConsistentQuotesInMultiline config param for Style/StringLiterals forces all literals which are concatenated using \ to use the same quote style
  - Disable Style/StringMethods by default
  - New cop Style/FrozenStringLiteralComment can check for and add the comment # frozen_string_literal: true to the top of files. This might help with upgrading to Ruby 3.0
  - New cop Style/MutableConstant checks for assignment of mutable objects to constants

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ gem "therubyracer", platforms: :ruby
 gem "uglifier", ">= 2.7.2"
 
 # views
-gem "active_link_to"
+gem "active_link_to", ">= 1.0.3"
```
Comment (timurvafin) [inline Gemfile]: @slavakisel why we need to define here version condition here, probably updated version in `Gemfile.lock` should be enough?

File: app/sanitizers/user/parameter_sanitizer.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class ParameterSanitizer < Devise::ParameterSanitizer
       email
       password
       password_confirmation
-    )
+    ).freeze
```
Comment (timurvafin) [inline app/sanitizers/user/parameter_sanitizer.rb]: What is the reason for this change?

Comment (slavakisel) [inline app/sanitizers/user/parameter_sanitizer.rb]: It is a recommendation from new cop `Style/MutableConstant` because it is actually possible to modify non-frozen constants in ruby. E.g.:

```ruby
class Foo
  DEFAULTS = [:a, :b]
end

default = Foo::DEFAULTS
default << :c
Foo::DEFAULTS #=> [:a, :b, :c]
```
